src/Telemetry/AzureDevOpsClient/AzureDevOpsHttpClientFactory.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Internal.AzureDevOps
+{
+    public class AzureDevOpsHttpClientFactory : IHttpClientFactory
+    {
+        private IEnumerable<DelegatingHandler> _delegatingHandlers;
+
+        public AzureDevOpsHttpClientFactory(IEnumerable<DelegatingHandler> delegatingHandlers = null)
+        {
+            _delegatingHandlers = delegatingHandlers;
+        }
+
+        public HttpClient CreateClient(string name)
+        {
+            return HttpClientFactory.Create(new HttpClientHandler { CheckCertificateRevocationList = true }, _delegatingHandlers.ToArray());
+        }
```

Comment:
If I'm not mistaken, we already have factories like this. Can you search for `CheckCertificateRevocationList` to see?

I am asking because there is nothing AzDO specific about this factory so wondering why it has this name

Reply:
I'm not seeing any factory implementations. I can move the factory to the Utility project and give it a universal name like `HttpClientFactoryWithDelegatingHandlers` or something similar

Reply:
On a second thought, this factory probably knows it wants to inject the retry handler? Maybe we can hardcode that here?

Reply:
I made the client collect all `AzureDevOpsDelegatingHandlers`, in case we want to add more of them in the future